### PR TITLE
Miller-Rabin Primality Test

### PR DESCRIPTION
**Pull request by 0xWOLAND**
Implemented Miller-Rabin Primality Test

### CODE REVIEW

**Comment by 0xWOLAND** [comment]
closes #13

File: src/core_crypto/prime.rs [hunk 1]
```diff
@@ -1,15 +1,49 @@
...
 
 // trait PrimalityTest<T> {
 //     fn is_prime(candidate: T) -> bool;
 // }
 
-// TODO (Jay): this is only a workaround. Add a propoer way to perform primality
-// tests.
 fn is_probably_prime(candidate: u64) -> bool {
-    num_bigint_dig::prime::probably_prime(&num_bigint_dig::BigUint::from(candidate), 0)
+    const K: usize = 40;
```
Comment (Janmajayamall) [inline src/core_crypto/prime.rs:10]: Can you add reference for the implementation?

File: src/core_crypto/prime.rs [hunk 2]
```diff
@@ -120,4 +154,52 @@ mod test {
             "Incorrect {N}^th root of unity: {root}^{N} != 1"
         );
     }
+
+    #[test]
+    fn test_is_prime() {
```
Comment (Janmajayamall) [inline src/core_crypto/prime.rs:159]: We will need stronger tests.

File: src/core_crypto/prime.rs (hunk 1 above)
Comment (Janmajayamall) [inline src/core_crypto/prime.rs:10]: Can you add reference for the implementation?

**Review by Janmajayamall — CHANGES_REQUESTED** [review]

**Pull request closed without merging**